1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,25 @@
-function setAlarm() {}
+//   const element = document.getElementById("timeRemaining");
+// setInterval(function() {element.innerHTML += "#number"}, 1000);
+
+function setAlarm() {
+  const inputField = document.getElementById("alarmSet");
+  let timer = inputField.value;
+
+  const heading = document.getElementById("timeRemaining");
+  heading.innerText = `Time Remaining: 00: ${timer}`;
+
+  const myInterval = setInterval(() => {
+    timer = timer -1;
+    heading.innerText = `Time Remaining: 00: ${timer}`;
+
+    if (timer === 0) {
+      clearInterval(myInterval);
+      playAlarm();
+    }
+  }, 1000);
+
+  clearInterval()
+}
```

Comment:
Really nice work here Saf!!! 🎉  🕺  🦦 

I know you did not write this, but I am unsure why the `// DO NOT EDIT BELOW HERE` is there, when it says to work on that function in the task? Strange...

Either way, here is what I did to get the stop to work

```
const heading = document.getElementById("timeRemaining");
const inputField = document.getElementById("alarmSet");
let myInterval;

function setAlarm() {
	let timer = inputField.value;
	clearInterval(myInterval);
	heading.innerText = `Time Remaining: 00: ${timer}`;

	myInterval = setInterval(() => {
		timer = timer -1;
		heading.innerText = `Time Remaining: 00: ${timer}`;

		if(timer === 0) {
			clearInterval(myInterval);
			playAlarm();
		}
	}, 1000);

	clearInterval();
}

function stopAlarm() {
	clearInterval(myInterval);
	heading.innerText = "Time Remaining: 00: 00";
	audio.pause();
}

let audio = new Audio("alarmsound.mp3");

function setup() {
	document.getElementById("set").addEventListener("click", () => {
		setAlarm();
	});

	document.getElementById("stop").addEventListener("click", () => {
		stopAlarm();
	});
}

function playAlarm() {
	audio.play();
}

window.onload = setup;
```

I always suggest if you are grabbing a HTML element and setting it as a vaiaible, ie `const heading = document.getElementById("timeRemaining");` you always do this at the very top of the file so it is at global scope and all the functions and as such can access it :) 

Again, great work!!!